### PR TITLE
Macport issues

### PR DESCRIPTION
**Pull request by jedwards4b**
Further cleanup and fixes addressing running scripts_regression_tests.py on a mac.
Issue #1644 is an outstanding problem with this port.

Test suite: scripts_regression_tests.py
Test baseline: 
Test namelist changes: 
Test status: bit for bit

Fixes 

User interface changes?: 

Code review:

### CODE REVIEW

File: config/acme/machines/Makefile [hunk 1]
```diff
@@ -90,12 +90,7 @@ ifeq ($(USE_ESMF_LIB), TRUE)
    CPPDEFS += -DUSE_ESMF_LIB
 endif
 
-# ESMF_INTERFACE is currently only defined in env_build.xml
```
Comment (jedwards4b) [inline config/acme/machines/Makefile]: This is just a clean-up - there is no longer an ESMF comp_interface

File: config/acme/machines/Makefile [hunk 2]
```diff
@@ -351,6 +346,7 @@ else
   endif
 endif
 
+COMP_INTERFACE := $(shell echo $(COMP_INTERFACE) | tr A-Z a-z)
```
Comment (jedwards4b) [inline config/acme/machines/Makefile]: COMP_INTERFACE is passed in uppercase, but needs to be lower case in the include path

File: scripts/lib/CIME/SystemTests/system_tests_common.py [hunk 3]
```diff
@@ -480,16 +480,13 @@ def build_phase(self, sharedlib_only=False, model_only=False):
 """
 echo Insta pass
 echo SUCCESSFUL TERMINATION > {}/cpl.log.$LID
-cp {}/scripts/tests/cpl.hi1.nc.test {}/{}.cpl.hi.0.nc.base
+cp {}/scripts/tests/cpl.hi1.nc.test {}/{}.cpl.hi.0.nc
```
Comment (jedwards4b) [inline scripts/lib/CIME/SystemTests/system_tests_common.py]: Without this change this test fails in the run phase with 
```
ERROR: copy failed: no hist files found in rundir '/Users/jedwards/projects/scratch/TESTTESTDIFF_P1.f19_g16_rx1.A.elm_gnu.20170606_074337_ml444i/run'
```
It's not clear to me why it isn't failing on other systems.

File: scripts/lib/CIME/SystemTests/system_tests_common.py [hunk 4]
```diff
@@ -480,16 +480,13 @@ def build_phase(self, sharedlib_only=False, model_only=False):
...
 cp {}/scripts/tests/cpl.hi2.nc.test {}/{}.cpl.hi.0.nc.rest
 """.format(rundir, cimeroot, rundir, case, cimeroot, rundir, case)
         self._set_script(script)
         super(TESTTESTDIFF, self).build_phase(sharedlib_only=sharedlib_only,
                                               model_only=model_only)
 
-    def run_indv(self, suffix=None, st_archive=False ):
```
Comment (jgfouca) [inline scripts/lib/CIME/SystemTests/system_tests_common.py]: @jedwards4b this is probably what caused the copy fail above (this run_indv changed its defaults). Looks like you found a way around it, so this is a good change.

**Pull request merged**